File: app/src/main/java/com/scaredeer/offlinedocument/MainActivity.java
package com.scaredeer.offlinedocument;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.view.ViewGroup;
import android.webkit.WebView;
import android.webkit.WebViewClient;

public class MainActivity extends AppCompatActivity {
    private static final String OFFLINE_DOCUMENT_ROOT = "file:///android_asset/offline_document";
    private static final String HOME_DOCUMENT = "index.html";
    private WebView mWebView;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        mWebView = new WebView(this);
        mWebView.setWebViewClient(new WebViewClient());
        mWebView.loadUrl(OFFLINE_DOCUMENT_ROOT + "/" + HOME_DOCUMENT);

        setContentView(mWebView);
    }

    @Override
    public void onBackPressed() {
        if (mWebView.canGoBack()) {
            mWebView.goBack();
        } else {
            super.onBackPressed();
        }
    }
}

File: app/src/main/kotlin/com/scaredeer/offlinedocument/MainActivity.kt
package com.scaredeer.offlinedocument

import android.content.res.Configuration
import android.os.Bundle
import android.webkit.WebView
import android.webkit.WebViewClient
import androidx.activity.OnBackPressedCallback
import androidx.appcompat.app.AppCompatActivity

private const val OFFLINE_DOCUMENT_ROOT = "file:///android_asset/offline_document"
private const val HOME_DOCUMENT = "index.html"

class MainActivity : AppCompatActivity() {

    private lateinit var webView: WebView

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        webView = WebView(this)
        webView.webViewClient = WebViewClient()
        webView.loadUrl("$OFFLINE_DOCUMENT_ROOT/$HOME_DOCUMENT")

        setContentView(webView)

        onBackPressedDispatcher.addCallback(this, object: OnBackPressedCallback(true) {
            override fun handleOnBackPressed() {
                if (webView.canGoBack()) {
                    webView.goBack()
                } else {
                    finish()
                }
            }
        })
    }

    /*
    cf. Manage WebView state
    https://developer.android.com/develop/ui/compose/quick-guides/content/manage-webview-state
     */
    override fun onConfigurationChanged(newConfig: Configuration) {
        super.onConfigurationChanged(newConfig)
        webView.invalidate()
    }
}